server/src/main/java/org/elasticsearch/common/joda/JodaDeprecationPatterns.java

```diff
@@ -0,0 +1,94 @@
+        JODA_PATTERNS_DEPRECATIONS.put("Y", "'Y' year-of-era should be replaced with 'y'. Use 'Y' for week-based-year.");
+        JODA_PATTERNS_DEPRECATIONS.put("y", "'y' year should be replaced with 'u'. Use 'y' for year-of-era.");
+        JODA_PATTERNS_DEPRECATIONS.put("C", "'C' century of era is no longer supported.");
+        JODA_PATTERNS_DEPRECATIONS.put("x", "'x' weak-year should be replaced with 'Y'. Use 'x' for zone-offset.");
+        JODA_PATTERNS_DEPRECATIONS.put("Z",
+            "'Z' time zone offset/id fails when parsing 'Z' for Zulu timezone. Consider using 'X'.");
+        JODA_PATTERNS_DEPRECATIONS.put("z",
+            "'z' time zone text. Will print 'Z' for Zulu given UTC timezone.");
+    }
+
+    /**
+     * Checks if date parsing pattern is deprecated.
+     * Deprecated here means: when it was not already prefixed with 8 (meaning already upgraded)
+     * and it is not a predefined pattern from <code>FormatNames</code>  like basic_date_time_no_millis
+     * and it uses pattern characters which changed meaning from joda to java like Y becomes y.
+     * @param pattern - a format to be checked
+     * @return true if format is deprecated, otherwise false
+     */
+    public static boolean isDeprecatedPattern(String pattern) {
+        List<String> patterns = DateFormatter.splitCombinedPatterns(pattern);
+
+        for (String subPattern : patterns) {
+            boolean isDeprecated = subPattern.startsWith("8") == false && FormatNames.exist(subPattern) == false &&
+                JODA_PATTERNS_DEPRECATIONS.keySet().stream()
+                                          .filter(s -> subPattern.contains(s))
+                                          .findAny()
+                                          .isPresent();
+            if (isDeprecated) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Formats deprecation message for suggestion field in a warning header.
+     * Joins all warnings in a one message.
+     * @param pattern - a pattern to be formatted
+     * @return a formatted deprecation message
+     */
+    public static String formatSuggestion(String pattern) {
+        List<String> patterns = DateFormatter.splitCombinedPatterns(pattern);
+
+        Set<String> warnings = new LinkedHashSet<>();
+        for (String subPattern : patterns) {
+            if (isDeprecatedPattern(subPattern)) {
+                String suggestion = JODA_PATTERNS_DEPRECATIONS.entrySet().stream()
+                                                              .filter(s -> subPattern.contains(s.getKey()))
+                                                              .map(s -> s.getValue())
+                                                              .collect(Collectors.joining("; "));
+                warnings.add(suggestion);
+            }
+        }
+        String combinedWarning = warnings.stream()
+                                 .collect(Collectors.joining(";"));
```

Comment:
Since this is building a message for the end user it would be nice to put a space after the semi-colon.

Reply:
good point, this would be more readable with a space